src/passes/OptimizeInstructions.cpp

```diff
@@ -990,17 +952,24 @@ struct OptimizeInstructions
           }
         }
       }
-      if (binary->op == OrInt32) {
+      if (binary->op == OrInt32 || binary->op == XorInt32) {
         // an or flowing into a boolean context can consider each input as
```

Comment:
The comment here should be updated for xor as well.

But why is this true for xor? I think it's not. For example,
```
if (!!1 xor !!2) { .. }
```
If we do `optimizeBoolean` on the xor's inputs, we'll get
```
if (1 xor 2) { .. }
```
which is different. Am I missing something?

Reply:
Hmm, good catch!